src/main/java/net/dv8tion/jda/api/entities/Entitlement.java

```diff
@@ -137,11 +138,58 @@ default String getGuildId()
     @Nullable
     OffsetDateTime getTimeEnding();
 
+    /**
+     * Whether the {@link Entitlement Entitlement} was consumed or not.
+     *
+     * @return True if the {@link Entitlement Entitlement} was consumed, False otherwise
+     */
+    boolean isConsumed();
+
+    /**
+     * Consumes the {@link Entitlement Entitlement} if it has not already been consumed.
```

Comment:
The description should explain what the consuming is for, explaining that its used for one-time use etc.